src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -316,8 +316,11 @@ void ScheduleTask(DateTimeOffset utcNow)
         private static void ScanForExpiredItems(MemoryCache cache)
         {
             DateTimeOffset now = cache._lastExpirationScan = cache._options.Clock.UtcNow;
-            foreach (CacheEntry entry in cache._entries.Values)
+
+            foreach (KeyValuePair<object, CacheEntry> item in cache._entries)
             {
+                CacheEntry entry = item.Value;
+
```

Comment:
Could just deconstruct it instead and use the discard on the key if it isn't used?

```suggestion
            foreach ((_, CacheEntry entry) in cache._entries)
            {
```

Reply:
This is built for [`netstandard2.0;net461`](https://github.com/dotnet/runtime/blob/f9300c6aae7434971ec07bf0a7312cc5b2d60217/src/libraries/Microsoft.Extensions.Caching.Memory/src/Microsoft.Extensions.Caching.Memory.csproj#L4). The [`KeyValuePair<TKey,TValue>.Deconstruct(TKey, TValue)`](https://docs.microsoft.com/dotnet/api/system.collections.generic.keyvaluepair-2.deconstruct?view=net-5.0#System_Collections_Generic_KeyValuePair_2_Deconstruct__0___1__) method does not exist in those target frameworks. Sure you could have an extension method but then it would not be shorter as a whole.